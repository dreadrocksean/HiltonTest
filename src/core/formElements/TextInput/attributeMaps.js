const inputMap = {
  username: {
    autoComplete: 'username',
    type: 'text',
  },
  password: {
    autoComplete: 'current-password',
    type: 'password',
  },
  email: {
    autoComplete: 'email',
    type: 'email',
  },
  name: {
    autoComplete: 'name',
    type: 'name',
  },
  phone: {
    autoComplete: 'tel',
    type: 'tel',
  },
  address: {
    autoComplete: 'street-address',
    type: 'street-address',
  },
  city: {
    autoComplete: 'address-level2',
    type: 'address-level2',
  },
  state: {
    autoComplete: 'address-level1',
    type: 'address-level1',
  },
  zip: {
    autoComplete: 'postal-code',
    type: 'postal-code',
  },
  date: {
    autoComplete: 'bday',
    type: 'date',
  },
  'datetime-local': {
    type: 'datetime-local',
  },
  time: {
    type: 'time',
  },
  week: {
    type: 'week',
  },
  month: {
    autoComplete: 'bday-month',
    type: 'month',
  },
  url: {
    autoComplete: 'url',
    type: 'url',
  },
  color: {
    type: 'color',
  },
  gender: {
    autoComplete: 'gender',
    type: 'text',
  },
};

export const formInputMap = type => {
  const formatType = inputMap[type];
  const name = type.replace(/\b\w/g, l => l.toUpperCase()) || '';
  if (!formatType) {
    return { type, name };
  }
  return { ...inputMap[type], name };
};
