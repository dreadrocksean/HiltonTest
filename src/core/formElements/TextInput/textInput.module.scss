@import '../../../layout/styles/default_settings';

.textInput {
    & + .textInput {
        margin-top: 1rem;
    }
    .validationHelp {
        color: $errorRed;
        display: flex;
        justify-content: flex-end;
        margin-top: 5px;
    }
}

/* Base form input wrapper */
.formInput {
    display: block;
    margin: 1em 0;
    position: relative; // Contain help menus that are absolute positioned.
}

/* Group of form input wrappers */
.formInputGroup {
    display: flex;
    margin: 1em -0.5em;

    .formInput {
        flex: 1 1 50%;
        margin: 0 0.5em;
    }
}

/* Placeholder for disabled input */
.formInputDisabled {
    font-weight: normal;
    margin: 0.5em 0 0;
}

/* Input helper text */
.formInputMeta {
    font-size: 0.8em;
    margin: 0.25em 0 1.5em;
    text-align: right;
    font-weight: normal;

    a {
        text-decoration: underline;
        color: $gray3;
    }
}

label {
    position: relative; // Contain child help menus that are absolute positioned.
    font-size: 0.9em;
    font-weight: bold;
    line-height: 1.2;
    display: block;
    margin: 0 0 0.2em;
    color: $black;

    /* When the label is the error wrapper (like for checkboxes) */
    &.error {
        padding: 0.5em;
    }
}

input {
    width: 100%;
    height: 3.75em;
    margin-top: 0.25em;
    padding: 0 1em;
    border: 1px solid $borderGray;
    border-radius: 0.3em;
    font-size: 1em;
    color: $gray1;
}

/* Default text color for inputs */
input:valid {
    color: $gray1;
}

/* Error color for inputs with invalid values as determined by the user's browser */
input:invalid {
    color: $errorRed;
}

/* Read-only inputs */
input[readonly],
input[readonly='readonly'] {
    color: $gray9;
}
