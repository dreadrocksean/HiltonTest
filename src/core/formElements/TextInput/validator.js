export const validatorMap = {
  username: /^.+$/,
  address: /a-zA-Z\d\s-,#.\+]+/,
  zip: {
    pattern: /^\d{5,6}(?:[-\s]\d{4})?$/,
    help: value => {
      if (value) return;
    },
  },
  password: /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[$@$!%*?\-&])[A-Za-z\d$@$!%*\-?&]{8,}/,
  phone: /^(?:(?:\+?1\s*(?:[.-]\s*)?)?(?:\(\s*([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9])\s*\)|([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9]))\s*(?:[.-]\s*)?)?([2-9]1[02-9]|[2-9][02-9]1|[2-9][02-9]{2})\s*(?:[.-]\s*)?([0-9]{4})(?:\s*(?:#|x\.?|ext\.?|extension)\s*(\d+))?$/,
};

export const validator = (value, pattern) => {
  if (!pattern) return true;
  return pattern.test(value);
};

export const nonRegexValidator = (value, pattern) => {
  if (!pattern) return true;
  return pattern.test(value);
};
