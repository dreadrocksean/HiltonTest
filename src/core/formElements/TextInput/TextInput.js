import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { validator, validatorMap } from './validator';
import { formInputMap } from './attributeMaps';

import styles from './textInput.module.scss';

class TextInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      validating: false,
    };
    this.handleBlur = this.handleBlur.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.renderValidationHelp = this.renderValidationHelp.bind(this);
  }

  handleBlur() {
    this.setState({ validating: true });
  }

  handleChange(e) {
    this.props.handleChange(e, this.props.type);
  }

  getName() {
    const inputFormat = formInputMap(this.props.type);
    return (this.props.name || inputFormat.name).replace(/\b\w/g, l => l.toUpperCase()) || '';
  }

  getValidationMsg() {
    return `Not a valid ${this.props.type}`;
  }

  renderValidationHelp() {
    let msg =
      this.props.validator !== false && !this.props.value
        ? "Can't be empty"
        : this.getValidationMsg();
    const pattern = this.props.validator || validatorMap[this.props.type];
    if (this.state.validating && !validator(this.props.value, pattern)) {
      return <div className={styles.validationHelp}>{msg}</div>;
    }
  }

  render() {
    const { value, hidden } = this.props;
    const { type, autoComplete, name } = formInputMap(this.props.type);
    const fieldName = (this.props.name || name).replace(/\b\w/g, l => l.toUpperCase()) || '';

    return (
      <div className={styles.textInput}>
        <label>
          {!hidden && name}
          <input
            type={type}
            name={fieldName}
            format={type}
            autoComplete={autoComplete}
            value={value}
            onChange={this.handleChange}
            onBlur={this.handleBlur}
            hidden={hidden}
          />
          {this.renderValidationHelp()}
        </label>
      </div>
    );
  }
}

TextInput.propTypes = {
  type: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  hidden: PropTypes.bool,
  name: PropTypes.string,
  validator: PropTypes.string,
};

export default TextInput;
