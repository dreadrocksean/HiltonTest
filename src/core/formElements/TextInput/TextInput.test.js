import React from 'react';
import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';
import TextInput from './TextInput';

const props = {
  type: 'text',
  handleChange: () => console.log('handling change'),
  value: 'Test Value',
};

it('renders without crashing', () => {
  shallow(<TextInput {...props} />);
});

describe('TextInput component renders correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<TextInput {...props} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});

test('Expect component to be called at least once', () => {
  const wrapper = shallow(<TextInput {...props} />);
  expect(wrapper.find('.textInput').length).toEqual(1);
});
