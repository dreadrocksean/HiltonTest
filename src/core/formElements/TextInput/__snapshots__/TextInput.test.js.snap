// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component renders correctly renders correctly 1`] = `
<div
  className="textInput"
>
  <label>
    Text
    <input
      format="text"
      name="Text"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="Test Value"
    />
  </label>
</div>
`;
